modules/internal/DefaultAssociative.chpl

```diff
@@ -995,8 +995,14 @@ module DefaultAssociative {
     }
     return hash;
   }
-  
+
+  // Non-nilable classes will coerce to this.
   inline proc chpl__defaultHash(o: borrowed object): uint {
+    return _gen_key(__primitive("object2int", o));
+  }
+
+  // Nilable classes will coerce to this.
+  inline proc chpl__defaultHash(o: borrowed object?): uint {
```

Comment:
Actually you should be able to just have this version and leave out the above since any class should coerce to `borrowed object?`. Could you give that a try?

Reply:
I should have thought about that. That should work nicely.